core/src/main/java/com/sequenceiq/cloudbreak/service/environment/EnvironmentService.java

```diff
@@ -142,9 +142,15 @@ protected void prepareDeletion(Environment environment) {
     }
 
     public DetailedEnvironmentResponse delete(String environmentName, Long workspaceId) {
-        Environment environment = getByNameForWorkspaceId(environmentName, workspaceId);
-        delete(environment);
-        return conversionService.convert(environment, DetailedEnvironmentResponse.class);
+        try {
+            return transactionService.required(() -> {
+                Environment environment = getByNameForWorkspaceId(environmentName, workspaceId);
+                delete(environment);
+                return conversionService.convert(environment, DetailedEnvironmentResponse.class);
```

Comment:
Would be better to create a simple converter, which doesn't load the half database for the result.

Reply:
ok